File: app/src/androidTest/java/com/abishov/hexocat/HexocatTestRunner.java
package com.abishov.hexocat;

import android.app.Application;
import android.app.Instrumentation;
import android.content.Context;
import android.os.Bundle;
import android.support.test.runner.AndroidJUnitRunner;
import com.squareup.rx2.idler.Rx2Idler;
import io.appflate.restmock.RESTMockFileParser;
import io.appflate.restmock.RESTMockServerStarter;
import io.appflate.restmock.android.AndroidAssetsFileParser;
import io.appflate.restmock.android.AndroidLogger;
import io.reactivex.plugins.RxJavaPlugins;

public final class HexocatTestRunner extends AndroidJUnitRunner {

  @Override
  public Application newApplication(ClassLoader cl, String className, Context context)
      throws InstantiationException, IllegalAccessException, ClassNotFoundException {
    return Instrumentation.newApplication(HexocatTestApp.class, context);
  }

  @Override
  public void onCreate(Bundle arguments) {
    super.onCreate(arguments);

    RESTMockFileParser parser = new AndroidAssetsFileParser(getContext());
    RESTMockServerStarter.startSync(parser, new AndroidLogger());
  }

  @Override
  public void onStart() {
    RxJavaPlugins.setInitComputationSchedulerHandler(
        Rx2Idler.create("rxjava-scheduler-computation"));
    RxJavaPlugins.setInitIoSchedulerHandler(
        Rx2Idler.create("rxjava-scheduler-io"));

    super.onStart();
  }
}

File: app/src/androidTest/java/com/abishov/hexocat/HexocatTestRunner.kt
package com.abishov.hexocat

import android.app.Application
import android.app.Instrumentation
import android.content.Context
import android.os.Bundle
import androidx.test.runner.AndroidJUnitRunner
import io.appflate.restmock.RESTMockOptions
import io.appflate.restmock.RESTMockServerStarter
import io.appflate.restmock.android.AndroidAssetsFileParser
import io.appflate.restmock.android.AndroidLogger

class HexocatTestRunner : AndroidJUnitRunner() {

  @Throws(
    InstantiationException::class,
    IllegalAccessException::class,
    ClassNotFoundException::class
  )
  override fun newApplication(cl: ClassLoader, className: String, context: Context): Application {
    return Instrumentation.newApplication(HexocatTestApp::class.java, context)
  }

  override fun onCreate(arguments: Bundle) {
    super.onCreate(arguments)

    val parser = AndroidAssetsFileParser(context)
    val options = RESTMockOptions.Builder()
      .useHttps(true)
      .build()

    RESTMockServerStarter.startSync(parser, AndroidLogger(), options)
  }
}
